### PR TITLE
fix issues reported by cppcheck

### PR DESCRIPTION
**Pull request by hzhuang1**
Run command "make clean cppcheck".

---- static analyzer - cppcheck ----
cppcheck . --force --enable=warning,portability,performance,style --error-exitcode=1 > /dev/null cli/xsum_os_specific.c:220:9: style: The scope of the variable 'ret' can be reduced. [variableScope]
    int ret;
        ^
cli/xsum_sanity_check.c:571:19: style: Condition 'len>0' is always true [knownConditionTrueFalse]
    } else if (len>0) {
                  ^
cli/xsum_sanity_check.c:569:13: note: Assuming that condition 'len==0' is not redundant
    if (len == 0) {
            ^
cli/xsum_sanity_check.c:571:19: note: Condition 'len>0' is always true
    } else if (len>0) {
                  ^
tests/bench/benchHash.c:112:55: error: Uninitialized variables: result.internal_never_ever_use_directly, result.error_result_never_ever_use_directly, result.error_tag_never_ever_use_directly [uninitvar]
    BMK_runTime_t const runTime = BMK_extract_runTime(result);
                                                      ^
tests/bench/benchHash.c:107:12: note: Assuming condition is false
    while (!BMK_isCompleted_TimedFn(txf)) {
           ^
tests/bench/benchHash.c:107:12: note: Assuming condition is false
    while (!BMK_isCompleted_TimedFn(txf)) {
           ^
tests/bench/benchHash.c:107:12: note: Assuming condition is false
    while (!BMK_isCompleted_TimedFn(txf)) {
           ^
tests/bench/benchHash.c:112:55: note: Uninitialized variables: result.internal_never_ever_use_directly, result.error_result_never_ever_use_directly, result.error_tag_never_ever_use_directly
    BMK_runTime_t const runTime = BMK_extract_runTime(result);
                                                      ^
tests/bench/benchfn.c:95:12: error: Uninitialized variable: outcome.error_result_never_ever_use_directly [uninitvar]
    return outcome;
           ^
tests/collisions/main.c:209:14: style: Same expression on both sides of '<='. [duplicateExpression]
    assert(m <= m);
             ^
make: *** [Makefile:387: cppcheck] Error 1

Signed-off-by: Haojian Zhuang <haojian.zhuang@linaro.org>

### CODE REVIEW

File: cli/xsum_os_specific.c [hunk 1]
```diff
@@ -235,6 +234,7 @@ static int XSUM_vasprintf(char** strp, const char* format, va_list ap)
         *strp = NULL;
         return size;
     } else {
+        int ret;
```
Comment (Cyan4973) [inline cli/xsum_os_specific.c:237]: 👍

File: tests/collisions/main.c [hunk 2]
```diff
@@ -206,7 +206,6 @@ static double Cnm(int n, int m)
 {
     assert(n > 0);
     assert(m > 0);
-    assert(m <= m);
```
Comment (Cyan4973) [inline tests/collisions/main.c:209]: I think the correct fix here is : 
```
assert(m <= n);
```

File: cli/xxhsum.c [hunk 3]
```diff
@@ -59,7 +59,8 @@
 
 static unsigned XSUM_isLittleEndian(void)
 {
-    const union { XSUM_U32 u; XSUM_U8 c[4]; } one = { 1 };   /* don't use static: performance detrimental  */
+    union { XSUM_U32 u; XSUM_U8 c[4]; } one;   /* don't use static: performance detrimental  */
```
Comment (Cyan4973) [inline cli/xxhsum.c]: You might want to check that the compiler still generates the same code for that variant.
Not concerned about correctness : it should be fine in both cases.
But we want to be sure that performance is also not impacted.

**Comment by hzhuang1** [comment]
OK. Let me check the instructions.

On 19 Oct 2022, at 14:38, Yann Collet ***@***.******@***.***>> wrote:



@Cyan4973 commented on this pull request.

________________________________

In cli/xxhsum.c<https://github.com/Cyan4973/xxHash/pull/746#discussion_r998997881>:

> @@ -59,7 +59,8 @@

 static unsigned XSUM_isLittleEndian(void)
 {
-    const union { XSUM_U32 u; XSUM_U8 c[4]; } one = { 1 };   /* don't use static: performance detrimental  */
+    union { XSUM_U32 u; XSUM_U8 c[4]; } one;   /* don't use static: performance detrimental  */


You might want to check that the compiler still generates the same code for that variant.
Not concerned about correctness : it should be fine in both cases.
But we want to be sure that performance is also not impacted.

**Comment by Cyan4973** [comment]
> OK. Let me check the instructions.

oh, nervermind.
I initially reacted thinking this endian detection function was in `xxhash.c`,
but it's in `xxhsum.c`,
and where it's employed, it's not performance critical.
So performance is not an issue.

I confused it for `XXH_isLittleEndian()`,
which is way more performance critical.

Interestingly, note that both `XXH_isLittleEndian()` and `XSUM_isLittleEndian()` are identical.
I presume this change is meant to please `cppcheck` static analyzer.
Since I don't see an equivalent change on `XXH_isLittleEndian()`, I presume it did not flag anything there.
It's strange that it seems to complain about `XSUM_isLittleEndian()`, but not `XXH_isLittleEndian()`,
given that both functions are identical...

**Comment by hzhuang1** [comment]
Let me remove the patch for unusedStructMember first.

**Review by Cyan4973 — APPROVED** [review]

**Pull request merged**